Add Android example code in Frame Drop Issue (#8153)
getsentry/sentry-docs: src/docs/product/issues/issue-details/performance-issues/frame-drop/index.mdx

Reviewer: Do you mind changing line 45 from 

```
## Example
```

to 

```
## Examples
```

since we have multiple now?

            val i = newList.indexOfFirst { it % 2 == 1 }
            newList.add(i.takeIf { it != -1 } ?: 0, n)
        } else {
            val i = newList.indexOfLast { it % 2 == 1 }
            newList.add(i.takeIf { it != -1 } ?: newList.size, n)
        }
    }
    return newList
}
```

Performance could be improved by moving the long computation off of the main thread to a coroutine:

```kotlin
// sort a list of numbers, with even numbers before the odd ones, then call onSorted with the sorted list as parameter.
fun sortEvenOdd(numbers: List<Int>, onSorted: (sorted: List<Int>) -> Unit): List<Int> {
    lifecycleScope.launch(Dispatchers.Default) {
    val newList = ArrayList<Int>()
        for (n in numbers) {
            if (n % 2 == 0) {
                val i = newList.indexOfFirst { it % 2 == 1 }
                newList.add(i.takeIf { it != -1 } ?: 0, n)
            } else {
                val i = newList.indexOfLast { it % 2 == 1 }
                newList.add(i.takeIf { it != -1 } ?: newList.size, n)
            }
        }
        withContext(Dispatchers.Main) {
            // Continue computation on the main thread
            onSorted(newList)
        }
    }
}
```

> There is no predefined way to fix performance of custom algorithms, due to their nature.

Reviewer: ```suggestion
There is no generally-applicable way to fix the performance of custom algorithms since each is different.
```

> In our example above, we were analyzing each number, then performed a search to find the right index to put the number, and finally inserting the item in the list, for every item on the list.

Reviewer: ```suggestion
In our example above, for every item on the list we analyze each number, perform a search to find the right index to put the number, and finally insert the item in the list.
```

> By using two separated lists for even and odd numbers, we can completely avoid the search, as the right index will always be the end of the list.
> Also, due to the nature of the ArrayList, inserting an element in the middle of the list is slower than inserting it at the end.

Reviewer: ```suggestion
By using two separate lists for even and odd numbers we can avoid the search step since the right index will always be the end of the list.
Also, due to the nature of the ArrayList, inserting an element in the middle of the list is slower than inserting it at the end.
```

> With the considerations above, the same function can be made faster:

Reviewer: ```suggestion
Given those considerations this function can be made faster:
```

```kotlin
// sort a list of numbers, with even numbers before the odd ones, then call onSorted with the sorted list as parameter.
fun sortEvenOddFaster(numbers: List<Int>, onSorted: (sorted: List<Int>) -> Unit): List<Int> {
    lifecycleScope.launch(Dispatchers.Default) {
        val evenList = ArrayList<Int>(numbers.size)
        val oddList = ArrayList<Int>(numbers.size)
        for (n in numbers) {
            if (n % 2 == 0) {
                evenList.add(n)
            } else {
                oddList.add(n)
            }
        }
        evenList.addAll(oddList)
        withContext(Dispatchers.Main) {
            // Continue computation on the main thread
            onSorted(evenList)
        }
    }
}
```

> There are several considerations to keep in mind to improve performance of an algorithm:
> 
> - You may be able to parallelize some work with the use of Threads (or Kotlin coroutines).
> - Explore and find the best data structures you can for your specific scope.
> - Some structures could be cached and reused throughout your function, reducing memory usage.
> - Evaluate different approaches to solve a problem.

Reviewer: ```suggestion
There are several considerations to keep in mind to improve the performance of an algorithm:

- You may be able to parallelize work by using threads (or Kotlin coroutines).
- Explore and find the best data structures you can for your specific scope.
- Some structures can be cached and reused throughout your function, reducing memory usage.
- Evaluate different approaches to solve a problem.
```
